module/os/freebsd/zfs/zfs_vnops_os.c

```diff
@@ -1624,7 +1624,8 @@ zfs_rmdir_(vnode_t *dvp, vnode_t *vp, const char *name, cred_t *cr)
 
 	dmu_tx_commit(tx);
 
-	cache_vop_rmdir(dvp, vp);
+	if (zfsvfs->z_use_namecache)
+		cache_vop_rmdir(dvp, vp);
```

Comment:
`else {
      cache_assert_no_entries(dvp);
      cache_assert_no_entries(vp);
}
`

you may need to ifdef this bit on version >= 14

Reply:
I think that this would be a good addition, but I'd like to fix the bug now and leave the enhancement for later.
So far I do not see any uses of `cache_assert_no_entries` in OpenZFS code, so maybe the asserts can be added systematically..

Reply:
I'm not *blocking* this, just suggesting. I don't think there is a problem to add, but whatever.